wpilibc/src/main/native/cpp/controller/LTVUnicycleController.cpp

```diff
@@ -54,52 +46,38 @@ void LTVUnicycleController::SetTolerance(const Pose2d& poseTolerance) {
 }
 
 ChassisSpeeds LTVUnicycleController::Calculate(
-    const Pose2d& currentPose, const Pose2d& poseRef,
+    const Pose2d& currentPose,
+    units::meters_per_second_t currentLinearVelocity,
+    const Pose2d& poseRef,
     units::meters_per_second_t linearVelocityRef,
     units::radians_per_second_t angularVelocityRef) {
-  Eigen::Matrix<double, 3, 1> x;
-  x(0, 0) = currentPose.Translation().X().to<double>();
-  x(1, 0) = currentPose.Translation().Y().to<double>();
-  x(2, 0) = currentPose.Rotation().Radians().to<double>();
+  m_poseError = poseRef.RelativeTo(currentPose);
 
-  Eigen::Matrix<double, 3, 1> r;
-  r(0, 0) = poseRef.Translation().X().to<double>();
-  r(1, 0) = poseRef.Translation().Y().to<double>();
-  r(2, 0) = poseRef.Rotation().Radians().to<double>();
+  if (currentLinearVelocity.to<double>() < 1e-9) {
```

Comment:
This should take the absolute value. Negative velocities less than -1e-9 are valid too and shouldn't be rounded up to 1e-9.